### DISCUSSION TITLE
How to know whether the API of Filament needs to run in the Filament thread?

### DISCUSSION THREAD

**Discussion by August1996** [google/filament#6833] [Q&A] [answered]
I used the main thread as the Filament thread for my Android App at first, but some APIs of Filament cost time. so I want to change the Filament thread from main thread to a worker thread, but I don't know what api must be used on the Filament thread and what interfaces It can be used on any thread.
@pixelflinger @bejado @poweifeng @romainguy  Could you give me some advices please?

**Comment by pixelflinger** [comment, marked as answer]
Filament must be used from a single thread. So you just move all filament calls (including creation) to the worker thread.

I'm interested to know which filament apis are taking excessive time though. You should be able to use filament from the main thread without too many issues.

**Reply by romainguy** [reply]
To emphasize what @pixelflinger said, you can use Filament from *any* thread, as long as it's always the same thread.

**Reply by August1996** [reply]
Ok, then I will move all my api calls to Filament thread, Thanks.
